### PR TITLE
Adds basic feature rule

### PR DESCRIPTION
**Pull request by nathaniel-brough**

### CODE REVIEW

**Comment by nathaniel-brough** [comment]
I think I've updated your role for this repository, so you should be able to "rebase and merge" beyond just approving. But let me know if you have any issues.

File: cc/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,20 @@
+load("@modular_cc_toolchain//cc:features.bzl", "cc_feature", "action_mux")
+
+load("@rules_cc//cc:action_names.bzl", "ALL_CC_COMPILE_ACTION_NAMES", "ALL_CC_LINK_ACTION_NAMES")
+
+cc_feature(
+    name = "garbage_collect_sections",
+    action_flags = action_mux({
+        tuple(ALL_CC_COMPILE_ACTION_NAMES): ["-ffunction-sections", "-fdata-sections"],
```
Comment (nathaniel-brough) [inline cc/BUILD.bazel]: There were a couple of quirks here, that I've just hacked around for now. Namely, the action group lists aren't hashable so I've just converted them to tuples for the moment. But the plan is to get this looking a little nicer.

**Review by nathaniel-brough — COMMENTED** [review]
I'm hoping to get a bit of a site up and running for docs. But I haven't landed on a solution just yet. I might go with mdbook, as it's what I'm most familiar with. But that's the general reason for all the extra targets under the /docs directory.

**Review by tpudlik — APPROVED** [review]
Looks good to me!

**Pull request merged**